accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,188 @@
+- `view` View registry info, also including **workspaces**
+
+#### Test example:
+
+```
+├── package.json { "name": "foo", "workspaces": ["dep-a", "dep-b"] }
+├── dep-a
+│   └── package.json { "version": "1.0.0" }
+└── dep-b
+    └── package.json { "version": "1.3.1" }
+
+$ npm test
+
+> foo@1.0.0 test /Users/username/foo
+> echo "Error: no test specified" && exit 1
+
+Error: no test specified
+npm ERR! Test failed.  See above for more details.
+
+> dep-a@1.0.0 test /Users/username/foo/dep-a
+> done
+
+> dep-b@1.3.1 test /Users/username/foo/dep-b
+> done
+```
+
+**NOTE:** `publish` and `version` are very special cases in a **npm workspaces** setup and are outside of the scope of this RFC. These features will live in userland in the form of popular modules such as [Lerna](https://lerna.js.org/) for the foreseeable future.
+
+### 2. Filter a subset of workspaces
+
+Rather than defining a strict UX, the current proposal is to adhere to a `workspace` command prefix in the form of either: `npm workspace <workspace_name> <command>` OR `npm ws <workspace_name> <command>` OR an even shorter shortcut using a symbol `npm :<workspace_name> <command>`, for example running `test` in a specific **workspace** named `foo` with these different ideas:
+
+- `npm workspace foo test`
+- `npm ws foo test`
+- `npm :foo test`
+
+Leaving space to gather feedback from the community on whether to introduce one of this proposed commands or even introducing some novel interface.
+
+#### Install dependency example:
+
+Add `tap` as a **dependency** of a **workspace** named `foo`:
+
+```sh
+npm workspace foo install tap
+npm ws foo install tap
+npm :foo install tap
+```
+
+### 3. Grouping/Aliasing workspaces
+
+One proposed feature that was surfaced during our brainstorm sessions is the ability for users to create arbitrary groups of **workspaces** defined via **workspaces configuration**, e.g:
+
+Given a **npm workspace** setup with the following file structure:
+
+```
+/root
+├── core
+│   ├── foo
+│   ├── bar
+│   └── util
+└── plugins
+    ├── helpers
+    ├── lorem
+    └── ipsum
+```
+
+It's possible to define a **groups** property within your **workspaces configuration** that will allow defining named sets of **workspaces** that can be later on used to run commands, e.g:
+
+```
+{
+    "name": "workspace-example",
+    "version": "1.0.0",
+    "workspaces": {
+        "groups": {
+            "core": ["core/*"], // accepts fs location
+            "plugins": ["lorem", "ipsum"], // also accepts workspaces names
+            "common": ["util", "helpers"]
+        },
+        "packages": [
+            "core/*",
+            "plugins/*"
+        ]
+    }
+}
+```
+
+Following up with the previous configuration example, it would be possible to run tests across all **workspaces** from the `plugins` group, as following:
+
+```
+// one of these:
+$ npm workspace plugins test
+$ npm ws plugins test
+$ npm :plugins test
```

Comment:
I am not a huge fan on this syntax.  It is mysterious and I would prefer either `workspace` or `ws`.

Reply:
On the other hand, I use `npm i` and `npm t` several dozen times a day, on average. Shorthands have their benefits (and they appear completely additive in this case).

Reply:
a long of "workspace" and a short of "ws" makes sense to me; making the short version do something different does not.